pkg/kn/commands/service/configuration_edit_flags.go

```diff
@@ -427,6 +432,28 @@ func (p *ConfigurationEditFlags) Apply(
 		servinglib.UpdateUser(template, p.PodSpecFlags.User)
 	}
 
+	if cmd.Flags().Changed("scale-init") {
+		contains := func(s []string, e string) bool {
+			for _, a := range s {
+				if strings.Contains(a, e) {
+					return true
+				}
+			}
+			return false
+		}
+
+		if cmd.Flags().Changed("annotation") && contains(p.Annotations, "autoscaling.knative.dev/initialScale") {
+			return fmt.Errorf("only one of the --scale-init or --annotation autoscaling.knative.dev/initialScale can be specified")
+		}
+		annotationsMap := make(map[string]string)
+		annotationsMap["autoscaling.knative.dev/initialScale"] = strconv.Itoa(p.ScaleInit)
```

Comment:
Is there a check for negative values of `ScaleInit`?

Reply:
All the validation should be done at the server side, I suppose. Even with the max and min scale option, there is no validation done on the client side.